src/TokenChecks.sol

```diff
@@ -0,0 +1,569 @@
+    function DOMAIN_SEPARATOR() external view returns (bytes32);
+    function nonces(address) external view returns (uint256);
+    function wards(address) external view returns (uint256);
+    function deny(address) external;
+    function approve(address, uint256) external returns (bool);
+    function transfer(address, uint256) external returns (bool);
+    function transferFrom(address, address, uint256) external returns (bool);
+    function increaseAllowance(address, uint256) external returns (bool);
+    function decreaseAllowance(address, uint256) external returns (bool);
+    function mint(address, uint256) external;
+    function burn(address, uint256) external;
+    function permit(address, address, uint256, uint256, bytes memory) external;
+    function permit(address, address, uint256, uint256, uint8, bytes32, bytes32) external;
+}
+
+interface SignerLike {
+    function isValidSignature(
+        bytes32,
+        bytes memory
+    ) external view returns (bytes4);
+}
+
+contract MockMultisig is SignerLike {
+    address public signer1;
+    address public signer2;
+
+    constructor(address signer1_, address signer2_) {
+        signer1 = signer1_;
+        signer2 = signer2_;
+    }
+
+    function isValidSignature(bytes32 digest, bytes memory signature) external view returns (bytes4 sig) {
+        bytes32 r;
+        bytes32 s;
+        uint8 v;
+        assembly {
+            r := mload(add(signature, 0x20))
+            s := mload(add(signature, 0x40))
+            v := byte(0, mload(add(signature, 0x60)))
+        }
+        if (signer1 == ecrecover(digest, v, r, s)) {
+            assembly {
+                r := mload(add(signature, 0x80))
+                s := mload(add(signature, 0xA0))
+                v := byte(0, mload(add(signature, 0xC0)))
+            }
+            if (signer2 == ecrecover(digest, v, r, s)) {
+                sig = SignerLike.isValidSignature.selector;
+            }
+        }
+    }
+}
+
+/** 
+ * @title Token Checks
+ * @dev Contains checks for token contracts similar to test/SampleToken.sol
+ */
+contract TokenChecks is DssTest {
+
+    using GodMode for *;
+
+    event Transfer(address indexed from, address indexed to, uint256 amount);
+    event Approval(address indexed owner, address indexed spender, uint256 amount);
+
+    function forceMint(address _token, address _who, uint256 _amount) internal {
+        uint256 prevWard = TokenLike(_token).wards(address(this));
+        _token.setWard(address(this), 1);
+        TokenLike(_token).mint(_who, _amount);
+        _token.setWard(address(this), prevWard);
+    }
+
+    // ************************************************************************************************************
+    // Mint/Burn
+    // ************************************************************************************************************
+
+    function checkMintBurn(address _token, string memory _contractName) public {
```

Comment:
If we leave this, maybe rename to `bulkCheckMintBurn`?
That will emphasize these checks also exist separately. 
Same for the other functions that group checks.

Reply:
Done